versioned_docs/version-V3/guides/providing-liquidity/setting-up-your-contract.md

```diff
@@ -0,0 +1,188 @@
+---
+id: setting-up
+title: Setting Up Your Contract
+sidebar_position: 1
+---
+
+## Setting up the Contract
```

Comment:
can we add an introduction paragraph to give the user some context? e.g.
- what is this guide about
- by the end of this guide, you will be able to do X. (or you will have a contract that does X)
- what Uniswap contracts does this interact with?

Reply:
I like this idea! And to help answer:
- This guide covers minting a position, adding liquidity to a position, decreasing liquidity, and collecting fees. It also is an example of a custodial contract for ERC721s.
- Should be able to interact with functions `increaseLiquidity` decreaseLiquidity`, `collect`, and `mint` from periphery.
- interacts with our periphery contracts, the main one being the INonFungiblePositionManager.sol

Reply:
I like this idea! And to help answer:
- This guide covers minting a position, adding liquidity to a position, decreasing liquidity, and collecting fees. It also is an example of a custodial contract for ERC721s.
- Should be able to interact with functions `increaseLiquidity` decreaseLiquidity`, `collect`, and `mint` from periphery.
- interacts with our periphery contracts, the main one being the INonFungiblePositionManager.sol